### PR TITLE
Convert to 2018 edition

### PR DESCRIPTION
**Pull request by djc**
I don't think compatibility with older compilers will be an issue for us.

### CODE REVIEW

File: quinn/examples/client.rs [hunk 1]
```diff
@@ -1,29 +1,24 @@
...
 use std::time::{Duration, Instant};
 
+use failure::Error;
 use futures::Future;
-use structopt::StructOpt;
+use quinn;
+use slog::{Drain, Logger};
+use slog_term;
+use structopt::{self, StructOpt};
+use tokio;
```
Comment (Ralith) [inline quinn/examples/client.rs]: Unnecessary crate use statements.

File: quinn/examples/server.rs [hunk 2]
```diff
@@ -18,22 +11,24 @@ use std::path::{self, Path, PathBuf};
...
-use structopt::StructOpt;
-use tokio::runtime::current_thread::Runtime;
-
-use failure::Error;
 use slog::{Drain, Logger};
+use slog_term;
+use structopt::{self, StructOpt};
+use tokio;
+use tokio::runtime::current_thread::Runtime;
+use tokio_current_thread;
```
Comment (Ralith) [inline quinn/examples/server.rs]: Unnecessary crate use statements.

File: quinn/src/lib.rs [hunk 3]
```diff
@@ -81,14 +68,17 @@ use futures::stream::FuturesUnordered;
...
 use slog::Logger;
 use tokio_io::{AsyncRead, AsyncWrite};
+use tokio_reactor;
 use tokio_timer::Delay;
 use tokio_udp::UdpSocket;
-
-use crate::quinn::{ConnectionHandle, Directionality, Side, StreamId};
+use untrusted;
+use webpki;
+use webpki_roots;
```
Comment (Ralith) [inline quinn/src/lib.rs]: Unnecessary crate use statements.

File: quinn-proto/src/lib.rs [hunk 4]
```diff
@@ -1,33 +1,24 @@
...
 extern crate lazy_static;
-extern crate orion;
-extern crate rand;
-extern crate ring;
-extern crate rustls;
-extern crate slab;
 #[macro_use]
 extern crate slog;
 #[cfg(test)]
 extern crate untrusted;
```
[the remark this answers is not in the document]
Comment (djc) [reply quinn-proto/src/lib.rs]: Yup, you're right! I had just used `cargo fix --edition-idioms` and made it cleanly work, so I didn't do much thinking. This means the code becomes even better. 👍

File: quinn-proto/src/endpoint.rs [hunk 5]
```diff
@@ -427,15 +427,15 @@ impl Endpoint {
...
             .encode(&mut buf);
-            {
-                let start = buf.len();
-                buf.resize(start + padding, 0);
-                self.ctx.rng.fill_bytes(&mut buf[start..start + padding]);
-            }
+
+            let start = buf.len();
+            buf.resize(start + padding, 0);
+            self.ctx.rng.fill_bytes(&mut buf[start..start + padding]);
```
[the remark this answers is not in the document]
Comment (djc) [reply quinn-proto/src/endpoint.rs]: In general all inner scopes that don't return a value look like NLL workarounds to me, so I do think the new version is more idiomatic -- okay to keep it?

File: quinn-proto/examples/connect.rs [hunk 6]
```diff
@@ -1,22 +1,19 @@
...
 use std::sync::Arc;
 use std::time::{Duration, Instant};
 
-use crate::quinn::{Config, Directionality, Endpoint, Event, Io, ReadError, Timer};
 use failure::Error;
+use quinn_proto::{self as quinn, Config, Directionality, Endpoint, Event, Io, ReadError, Timer};
 use rustls::ProtocolVersion;
 use slog::{Drain, Logger};
+use slog_term;
+use webpki_roots;
```
[the remark this answers is not in the document]
Comment (djc) [reply quinn-proto/examples/connect.rs]: Yup.

File: quinn-proto/src/lib.rs [hunk 7]
```diff
@@ -1,33 +1,24 @@
...
-extern crate webpki;
 
 use std::fmt;
 use std::ops;
 
+use bytes;
```
[the remark this answers is not in the document]
Comment (djc) [reply quinn-proto/src/lib.rs]: Check.

File: quinn-proto/src/tests.rs [hunk 8]
```diff
@@ -78,18 +78,13 @@ impl Default for Pair {
...
+    let certs = pemfile::certs(&mut reader).expect("cannot read certificates");
 
-    let keys = {
-        let f = fs::File::open("../certs/server.rsa").expect("cannot open '../certs/server.rsa'");
-        let mut reader = io::BufReader::new(f);
-        pemfile::rsa_private_keys(&mut reader).expect("cannot read private keys")
-    };
+    let f = fs::File::open("../certs/server.rsa").expect("cannot open '../certs/server.rsa'");
+    let mut reader = io::BufReader::new(f);
+    let keys = pemfile::rsa_private_keys(&mut reader).expect("cannot read private keys");
```
[the remark this answers is not in the document]
Comment (djc) [reply quinn-proto/src/tests.rs]: Hmm, yeah. I've reverted these.

File: quinn-proto/src/endpoint.rs (hunk 5 above)
[the remark this answers is not in the document]
Comment (djc) [reply quinn-proto/src/endpoint.rs]: Renamed to `padding_start`.

File: quinn/src/lib.rs [hunk 9]
```diff
@@ -81,15 +68,14 @@ use futures::stream::FuturesUnordered;
 use futures::task::{self, Task};
 use futures::unsync::oneshot;
 use futures::Stream as FuturesStream;
-use futures::{Async, Future, Poll, Sink};
+use futures::{self, Async, Future, Poll, Sink};
```
Comment (Ralith) [inline quinn/src/lib.rs]: I think `self` is redundant here

Comment (djc) [reply quinn/src/lib.rs]: Fixed.

**Comment by djc** [comment]
Needed another formatting fix, please reapprove?

**Pull request merged**